app/src/main/java/org/simple/clinic/bp/entry/ConfirmRemoveBloodPressureDialog.kt

```diff
@@ -39,17 +53,60 @@ class ConfirmRemoveBloodPressureDialog : AppCompatDialogFragment() {
     }
   }
 
+  @Inject
+  lateinit var controller: ConfirmRemoveBloodPressureDialogController
+
+  private val screenDestroys = PublishSubject.create<ScreenDestroyed>()
+  private val onStarts = PublishSubject.create<Any>()
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     TheActivity.component.inject(this)
   }
 
+  @SuppressLint("CheckResult")
   override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
-    return AlertDialog.Builder(requireContext(), R.style.Clinic_V2_DialogStyle_Destructive)
+    val dialog = AlertDialog.Builder(requireContext(), R.style.Clinic_V2_DialogStyle_Destructive)
         .setTitle(R.string.bloodpressureentry_remove_bp_title)
         .setMessage(R.string.bloodpressureentry_remove_bp_message)
         .setPositiveButton(R.string.bloodpressureentry_remove_bp_confirm, null)
         .setNegativeButton(R.string.bloodpressureentry_remove_bp_cancel, null)
         .create()
+
+    onStarts
+        .take(1)
+        .flatMap { setupDialog() }
+        .takeUntil(screenDestroys)
+        .subscribe { uiChange -> uiChange(this) }
+
+    return dialog
+  }
+
+  private fun setupDialog(): Observable<UiChange> {
+    return Observable.merge(dialogCreates(), screenDestroys, removeClicks())
+        .observeOn(io())
+        .compose(controller)
+        .observeOn(mainThread())
+  }
+
+  private fun removeClicks(): Observable<UiEvent> {
+    val button = (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+
+    return RxView.clicks(button).map { ConfirmRemoveBloodPressureDialogRemoveClicked }
+  }
+
+  private fun dialogCreates(): Observable<UiEvent> {
+    val bloodPressureMeasurementUuid = arguments!!.getSerializable(KEY_BP_UUID) as UUID
+    return Observable.just(ConfirmRemoveBloodPressureDialogCreated(bloodPressureMeasurementUuid))
+  }
+
+  override fun onStart() {
+    super.onStart()
+    onStarts.onNext(Any())
+  }
+
+  override fun onDestroyView() {
+    super.onDestroyView()
+    screenDestroys.onNext(ScreenDestroyed())
```

Comment:
Wanna just use `RxView.detaches(view)` instead of maintaining this extra Subject?